core/src/main/java/hudson/model/AsyncAperiodicWork.java

```diff
@@ -40,8 +40,48 @@
  * @since 1.410
  */
 public abstract class AsyncAperiodicWork extends AperiodicWork {
-	
-	/**
+    /**
+     * The default number of minutes after which to try and rotate the log file used by {@link #createListener()}.
+     * This value is controlled by the system property {@code hudson.model.AsyncAperiodicWork.logRotateMinutes}.
+     * Each individual AsyncAperiodicWork can also have a per-extension override using the system property
+     * based on their fully qualified class name with {@code .logRotateMinutes} appended.
+     *
+     * @since 1.650
+     */
+    private static final long LOG_ROTATE_MINUTES = Long.getLong(AsyncAperiodicWork.class.getName() +
+            ".logRotateMinutes", TimeUnit.DAYS.toMinutes(1));
+    /**
+     * The default file size after which to try and rotate the log file used by {@link #createListener()}.
+     * A value of {@code -1L} disables rotation based on file size.
+     * This value is controlled by the system property {@code hudson.model.AsyncAperiodicWork.logRotateSize}.
+     * Each individual AsyncAperiodicWork can also have a per-extension override using the system property
+     * based on their fully qualified class name with {@code .logRotateSize} appended.
+     *
+     * @since 1.650
```

Comment:
> then the version number in master should have been adjusted accordingly.

That would lead to incorrect versions already committed in git's master. You will have versions mix in git history that is wrong from maven versioning viewpoint.

> is a bad pattern as there are countless examples of them not actually being corrected for the merge.

That are release management issues and not PR author. This should be adjusted either before release, either during merge (gatekeeper/validated). That's why i suggest not using 'merge' button at all for gatekeepering with github-pullrequest-plugin (only labels and comments for triggering /o/ merges where you can enforce additional checks).